### ISSUE TITLE
[HL / TFC beta] VGUI - some menu entries are only partially displayed

### ISSUE THREAD

**Issue by pizzahut2** [ValveSoftware/halflife#2608]
Not sure if this happens in other Goldsource games as well, anyway some menu entries in the engineer menu are cut off in the beta.

This is what it should look like:
https://steamuserimages-a.akamaihd.net/ugc/798744098474196965/1F33F59CB83E63E68306E239A0E956D7286C409D/

This is what I get instead:
https://steamuserimages-a.akamaihd.net/ugc/798744098474197624/95A5CD1932A217FB6898A50D9E380FA317A1030C/

https://i.imgur.com/JPsdXAg.png

I've got a customised commandmenu.txt file, but I guess that shouldn't matter. Here's the relevant excerpt:

```
// ENGINEER
CUSTOM  "7" "BUILD"	"!BUILD"
{
	// SENTRY GUN
	CUSTOM		"1" "BUILD SENTRY GUN"		"!BUILDSENTRY"
	CUSTOM		"1" "DISMANTLE SENTRY"		"!DISMANTLESENTRY"
	CUSTOM		"2" "ROTATE SENTRY left"	"!ROTATESENTRY"
	CUSTOM		"3" "ROTATE SENTRY 180"		"!ROTATESENTRY180"
	ENGINEER	"4" "ROTATE SENTRY right"	"rotatesentry180;rotatesentry;rotatesentry;rotatesentry"
	CUSTOM		"0" "DETONATE SENTRY"		"!DETONATESENTRY"

	// DISPENSER
	CUSTOM		"5" "BUILD DISPENSER"		"!BUILDDISPENSER"
	CUSTOM		"5" "DISMANTLE DISPENSER"	"!DISMANTLEDISPENSER"

	// TELEPORTER
	"7" "TELEPORTER"
	{
```
PS: Did a search and didn't find something similar, still maybe someone else reported it already.

**Comment by mikela-valve** [comment]
Nice job catching that @pizzahut2! That was caused by the change that I made in #2387 to properly localize those GUI labels when they're actually localization tokens and not plain text. The localization function had a bug in it that caused strings larger than half the size of the buffer (32 characters) to be cut off early, so it wasn't show up for the majority of the labels but was especially apparent in your customized ones since they're significantly longer than the other labels.
